### PR TITLE
Labels: Added complex filtering logic

### PR DESCRIPTION
**Pull request by ebeers-png**
#### What's this PR do?

Adds more complex label filtering logic to the inventory list, by allowing users to filter the inventory by label on AND, OR, and EXCLUDE all at once. 

The complex label logic can be saved to filter groups, and migrations will update pre-existing filter groups to the new setup. 

#### Remaining work

Before merging this into develop, the inventory list menu should be re-organized to make room for the two new rows, which take up a lot of screen space.

The inventory map page, which also uses label filtering, was left unchanged, since it doesn't (yet) use Filter Groups and will work fine without the complex label logic. But it can be easily updated if needed, since its own code is already based on the inventory list's.

#### What are the relevant tickets?
#3371

#### Screenshots (if appropriate)

(Ignore the missing icons)

image

### CODE REVIEW

File: seed/migrations/0212_move_filtergroup_labels.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Generated by Django 3.2.18 on 2023-12-19 16:15
```
Comment (haneslinger) [inline seed/migrations/0212_move_filtergroup_labels.py:1]: 💯

**Review by haneslinger — COMMENTED** [review]
This is great! thank you! I have only a few notes, and a tiny commit that adds a test I would like to see passing. I'm happy to address them, if you like.

@kflemin, how to you want to handle that white space in the ui @ebeers-png mentioned above?

@ebeers-png, Have you noticed any different in query speed?

**Comment by ebeers-png** [comment]
@haneslinger Thank you for the review and suggestions! Added the catch for erroneous data to the update method, and I was able to simplify _get_label_views() a little more as well. 

Re: query speed, I haven't really noticed a difference in inventory load times. But that's mainly because SEED calculates included/excluded building ids on the frontend, in inventory_list_controller (lines 1110-1128). That's not the case for the custom report insights though, which uses _get_label_views() in models/data_views to get the right buildings. So I'd have to check whether the old code (with `set.intersection(...)` everywhere) is faster than what you suggested (though I doubt it).

**Comment by haneslinger** [comment]
The failing test is not this PRs fault. I'm addressing it now

**Pull request merged**